libs/librepcb/projecteditor/boardeditor/fsm/bes_select.cpp

```diff
@@ -705,6 +574,79 @@ BES_Base::ProcRetVal BES_Select::processSubStateMovingSceneEvent(
   return PassToParentState;
 }
 
+void BES_Select::addActionRotate(QMenu& menu, QString text) noexcept {
+  QAction* action = menu.addAction(QIcon(":/img/actions/rotate_left.png"), text);
+  connect(action, &QAction::triggered, [this](){
+    rotateSelectedItems(Angle::deg90());
+  });
+}
+
+void BES_Select::addActionFlip(QMenu& menu, QString text) noexcept {
+  QAction* action =  menu.addAction(QIcon(":/img/actions/flip_horizontal.png"), text);
+  connect(action, &QAction::triggered, [this](){
+    flipSelectedItems(Qt::Horizontal);
+  });
+}
+
+void BES_Select::addActionDelete(QMenu& menu, QString text) noexcept {
+  QAction* action = menu.addAction(QIcon(":/img/actions/delete.png"), text);
+  connect(action, &QAction::triggered, [this]() {
+    removeSelectedItems();
+  });
+}
+
+void BES_Select::addActionDeleteAll(QMenu& menu, BI_NetSegment* netsegment,
+                                        QString text) noexcept {
+  QAction* action = menu.addAction(QIcon(":/img/actions/minus.png"), text);
+  connect(action, &QAction::triggered, [this, netsegment](){
+    netsegment->setSelected(true);
+    removeSelectedItems();
+  });
+}
+
+void BES_Select::addActionMeasure(QMenu& menu, BI_NetLine* netline,
+                                      QString text) noexcept {
+  QAction* action = menu.addAction(QIcon(":/img/actions/ruler.png"), text);
+  connect(action, &QAction::triggered, [this, netline](){
+    netline->setSelected(true);
+    measureSelectedItems(*netline);
+  });
+}
+
+void BES_Select::addActionProperties(QMenu& menu, Board* board, BI_Base* item,
+                                     QString text) noexcept {
+  QAction* action = menu.addAction(QIcon(":/img/actions/settings.png"), text);
+  connect(action, &QAction::triggered, [this, board, item](){
+    openPropertiesDialog(*board, item);
+  });
+}
+
+void BES_Select::addActionSnap(QMenu& menu, const Point pos, Board* board,
+                                   BI_Base* item, QString text) noexcept {
+  if (!pos.isOnGrid(board->getGridProperties().getInterval())) {
+    QAction* action = menu.addAction(QIcon(":/img/actions/grid.png"), text);
+    connect(action, &QAction::triggered, [this, board, item](){
+      try  {
+        QScopedPointer<CmdDragSelectedBoardItems> cmdMove(
+              new CmdDragSelectedBoardItems(*board, item->getPosition()));
+        cmdMove->setCurrentPosition(item->getPosition().mappedToGrid(
+                                board->getGridProperties().getInterval()), false);
+        mUndoStack.execCmd(cmdMove.take());
+      } catch (Exception& e) {
+        QMessageBox::critical(&mEditor, tr("Error"), e.getMsg());
+      }
+    });
+  }
+}
+
+void BES_Select::addActionSelectAll(QMenu& menu, BI_NetSegment* netsegment,
```

Comment:
Since `netsegment` is not allowed to be `nullptr`, please pass it as a reference to make callers clear that it must not be null.

```suggestion
void BES_Select::addActionSelectAll(QMenu& menu, BI_NetSegment& netsegment,
```

The same also applies to arguments of other `addAction*` methods.